pkg/middleware/oapi_validate_test.go

```diff
@@ -87,6 +87,15 @@ paths:
       responses:
         '204':
           description: no content
+  /protected_resource_401:
+    get:
+      operationId: getProtectedResource
+      security:
+        - BearerAuth:
+          - unauthorized
```

Comment:
It's kind of odd to have an authorization scope named "unauthorized" which returns a 401, because you still have to have that scope present in your credentials. I see how it fits into the test, though, but this isn't the common way you'd do unauthorized errors.

Reply:
I just named it this way for the tests.

I hope the point of the functionality was clear.  Clients should get a 401 (e.g. with a www-authenticate header) if credentials are absent so they can retry with appropriate credentials.  The previous behavior of returning a 403 meant re-authenticating would make no difference.